lib/logstash/outputs/elasticsearch.rb

```diff
@@ -367,36 +367,37 @@ def multi_receive(events)
     wait_for_successful_connection if @after_successful_connection_done
     events_mapped = safe_interpolation_map_events(events)
     retrying_submit(events_mapped.successful_events)
-    unless events_mapped.failed_events.empty?
-      @logger.error("Can't map some events, needs to be handled by DLQ #{events_mapped.failed_events}")
-      send_failed_resolutions_to_dlq(events_mapped.failed_events)
+    unless events_mapped.event_mapping_errors.empty?
+      handle_event_mapping_errors(events_mapped.event_mapping_errors)
     end
   end
 
-  # @param: Arrays of EventActionTuple
+  # @param: Arrays of EventActionTuple with error messages
   private
-  def send_failed_resolutions_to_dlq(failed_action_tuples)
-    failed_action_tuples.each do |action|
-      handle_dlq_status(action, "warn", "Could not resolve dynamic index")
+  def handle_event_mapping_errors(event_mapping_errors)
+    event_mapping_errors.each do |action, message|
+      detail_message = message + ", " + action.to_s
+      handle_dlq_status(action.event, :warn, detail_message)
```

Comment:
I would prefer a rephrase of this, so that `event_mapping_errors` was _either_ an array of _`event`_/`message` tuples or an array of objects that have `event`/`message` properties, since `action` in this code is somewhat ambiguous and can mean either a String action or an action-tuple as prepared for a bulk request.

When we are building this array, we already have access to the event and an object that _has_ the message (the exception), and when we are raising the exception we have access to the `event`. We could either create a specialized struct, or can embed the event into the exception that we are already handling.

Additionally, we have everything we need to produce our `detail_message` when populating the array (or even when throwing the relevant exception that leads to an entry in the array), so I would rather pull that up.